src/controller/python/chip/clusters/Attribute.py

```diff
@@ -582,7 +582,7 @@ def SetErrorCallback(self, callback: Callable[[int, SubscriptionTransaction], No
             self._onErrorCb = callback
 
     @property
-    def OnAttributeChangeCb(self) -> Callable[[TypedAttributePath, SubscriptionTransaction], None]:
+    def OnAttributeChangeCb(self) -> Callable[[Union[TypedAttributePath, AttributePath], SubscriptionTransaction], None]:
```

Comment:
needs changes when setting _onAttributeChangeCb as well (like SetAttributeUpdateCallback) and after that followup that all usages can handle the different types.

We already seem to do quite badly here: https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_SC_3_6.py#L43 for example says it only supports a TypedAttributePath.

Overall:

```
❯ rg SetAttributeUpdateCallback --hidden --no-ignore --files-with-matches
src/controller/python/test/test_scripts/base.py
src/controller/python/test/test_scripts/cluster_objects.py
src/controller/python/chip/yaml/runner.py
src/controller/python/chip/clusters/Attribute.py
src/controller/python/chip/ChipDeviceCtrl.py
src/python_testing/TestGroupTableReports.py
src/python_testing/TC_SC_3_6.py
src/python_testing/TC_RR_1_1.py
src/python_testing/TC_ACE_1_2.py
```

Reply:
Looking more into this, just the raw path isn't helping much usually, we also need a way to access the (new) attribute data. Usually this is done via `GetAttribute()` or `GetAttributes()`, but this returns the typed attribute cache. So we would need a way to get untyped attribute cache as well.

Also, it starts to look really weird when we have to sprinkle if's in any OnAttributeChangeCb callback implementation. This goes a bit into API design, but I don't think we should use typing for this. Maybe a better approach would be to create a new `OnTypelessAttributeChangeCb` or `OnUntypedAttributeChangeCb`? The question is would we want to call that callback in any case or just when there is no type? :thinking: